src/Umbraco.Core/CryptoServiceProviderExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Security.Cryptography;
+
+namespace Umbraco.Core
+{
+    public static class CryptoServiceProviderExtensions
+    {
+        /// <summary>
+        /// Generates a random int withing a specified range.
+        /// </summary>
+        /// <param name="provider">Random bytes provider.</param>
+        /// <param name="minValue">The minimum value of the resulting int.</param>
+        /// <param name="maxValue">The maximum value of the resulting int.</param>
+        /// <returns>A random integer that falls withing the specified range.</returns>
+        public static int GetInt32(this RNGCryptoServiceProvider provider, int minValue, int maxValue)
+        {
+            var randomBytes = new byte[4];
+            provider.GetBytes(randomBytes);
+
+            var randomInt = Math.Abs(BitConverter.ToInt32(randomBytes, 0));
+            // We call do mod to ensure that the value is within the specified range.
+            return randomInt % (maxValue - minValue + 1) + minValue;
```

Comment:
I'm not sure this is what we want considering that this will generate a completely random integer in the entire int range, this would mean that if you had a smaller range, let's say 0-10, then the number is much more likely to be 10, than any other number, since the majority of the numbers in the int range is greater than 10 😄 

That's the benefit with using the modulus operator, this way we don't loose the randomness of the number, while essentially clamping it to 10 :) 

However it might be a good idea to extract the current clamping into its own method like you suggest :D

Reply:
Ahh okay, makes sense. So with a range on 0-10 and numbers like 24255, 46536, 35353, 65645 (completely random by typing my keyboard 😁 ), it won't be 10 always, but could be 3, 6, 9 as long it is within the range 😄

Reply:
Yes exactly 😄